server/src/main/java/org/apache/iotdb/db/engine/storagegroup/VirtualStorageGroupProcessor.java

```diff
@@ -1555,6 +1556,72 @@ public void timedFlushSeqMemTable() {
     }
   }
 
+  /** iterate over TsFiles and migrate to targetDir if out of ttl */
+  public void checkMigration(MigrationTask task) {
```

Comment:
Is it possible to move checkMigration and checkMigrateFile to MigrationTask?

Reply:
Currently i dont think a way to access the all virtual storage group processor in StorageGroupManager.